File: [23.12]Design-Pattern/Java/src/behavioral/memento/Adventurer.java
package behavioral.memento;

public class Adventurer {
    String job;
    int level;

    public Adventurer(String job, int level) {
        this.job = job;
        this.level = level;
    }

    public Memento createMemento(){
        return new Memento(job, level);
    }

    public void setInfo(Memento memento){
        this.job = memento.job;
        this.level = memento.level;
    }

    @Override
    public String toString() {
        return "Adventurer{" +
                "job='" + job + '\'' +
                ", level=" + level +
                '}';
    }
}

File: [23.12]Design-Pattern/Kotlin/src/main/kotlin/behavioral/memento/Adventurer.kt
package behavioral.memento

class Adventurer(
    var job: String,
    var level: Int
) {

    fun setData(memento: Memento) {
        this.job = memento.job
        this.level = memento.level
    }

    fun createMemento(): Memento{
        return Memento(job,level)
    }

    override fun toString(): String {
        return "Adventurer(job='$job', level=$level)"
    }


}